In [1]:
from model_selection import *

In [2]:
df1, df2 = load_data()
df1, df2 = resample_data(df1, df2, t='5S')
x_train, x_test, y_train, y_test = get_splitted_data(df1, df2)
x_train_std, x_test_std = get_scaled_data(x_train, x_test)

............... Datasets loaded.
............... Resampling completed.


In [3]:
x_train.shape, x_test.shape, y_train.shape, y_test.shape

((372750, 3), (159750, 3), (372750,), (159750,))

In [8]:
get_log_regr_best(x_train_std, y_train, x_test_std, y_test)

Time:  0:00:05.946340
Best params:  {'C': 1, 'penalty': 'l2'}
Best score: 0.585
-------------- Classification report -------------
              precision    recall  f1-score   support

           0       0.58      0.54      0.56     15660
           1       0.58      0.62      0.60     16290

    accuracy                           0.58     31950
   macro avg       0.58      0.58      0.58     31950
weighted avg       0.58      0.58      0.58     31950

---------------- Confusion matrix ----------------
[[ 8391  7269]
 [ 6109 10181]]
----------------
 Accuracy test data: 0.581


In [2]:
df1, df2 = load_data()
df1, df2 = resample_data(df1, df2, t='10S')
x_train_svm, x_test_svm, y_train_svm, y_test_svm = get_splitted_data(df1, df2)
x_train_svm, x_test_svm = get_scaled_data(x_train_svm, x_test_svm)
print('------------------------------------')
get_svm_best(x_train_svm, y_train_svm, x_test_svm, y_test_svm)

............... Datasets loaded.
............... Resampling completed.
------------------------------------
Time:  1:00:15.506155
Best params:  {'C': 1000, 'kernel': 'rbf'}
Best score: 0.761
-------------- Classification report -------------
              precision    recall  f1-score   support

           0       0.78      0.72      0.75      7870
           1       0.75      0.80      0.77      8105

    accuracy                           0.76     15975
   macro avg       0.76      0.76      0.76     15975
weighted avg       0.76      0.76      0.76     15975

---------------- Confusion matrix ----------------
[[5640 2230]
 [1589 6516]]
----------------

Accuracy test data: 0.761


In [5]:
get_tree_best(x_train, y_train, x_test, y_test)

Time:  0:01:37.160557
Best params:  {'max_depth': 15, 'min_samples_split': 2}
Best score: 0.828
-------------- Classification report -------------
              precision    recall  f1-score   support

           0       0.84      0.80      0.82     78304
           1       0.82      0.85      0.84     81446

    accuracy                           0.83    159750
   macro avg       0.83      0.83      0.83    159750
weighted avg       0.83      0.83      0.83    159750

---------------- Confusion matrix ----------------
[[62898 15406]
 [11812 69634]]
----------------
 Accuracy test data: 0.830


In [10]:
get_forrest_best(x_train, y_train, x_test, y_test)

Time:  0:27:13.639657
Best params:  {'max_depth': 15, 'n_estimators': 100}
Best score: 0.841
-------------- Classification report -------------
              precision    recall  f1-score   support

           0       0.86      0.81      0.83     15660
           1       0.83      0.87      0.85     16290

    accuracy                           0.84     31950
   macro avg       0.84      0.84      0.84     31950
weighted avg       0.84      0.84      0.84     31950

---------------- Confusion matrix ----------------
[[12691  2969]
 [ 2049 14241]]
----------------
 Accuracy test data: 0.843


In [11]:
get_gb_best(x_train, y_train, x_test, y_test)

n_estimators: 200.000000: accuracy = 0.78063045 | time elapsed:  0:04:59.716121
n_estimators: 300.000000: accuracy = 0.79403085 | time elapsed:  0:07:27.227580
n_estimators: 400.000000: accuracy = 0.79973172 | time elapsed:  0:09:23.771242


In [3]:
get_gb_best(x_train, y_train, x_test, y_test)

n_estimators: 100.000000: accuracy = 0.75656606 | time elapsed:  0:01:15.468095
n_estimators: 300.000000: accuracy = 0.79223340 | time elapsed:  0:04:59.733102
n_estimators: 500.000000: accuracy = 0.80355466 | time elapsed:  0:13:46.589515

--------------- Classification report -------------
              precision    recall  f1-score   support

           0       0.81      0.78      0.80     15556
           1       0.80      0.83      0.81     16394

    accuracy                           0.81     31950
   macro avg       0.81      0.81      0.81     31950
weighted avg       0.81      0.81      0.81     31950

----------------- Confusion matrix -----------------
[[12199  3357]
 [ 2817 13577]]
----------------------------------------------------

Accuracy test data: 0.807


Из ансамбевых классификаторов наиболее приемлемый результат продемонстрировала модель случайного леса на 100 деревьях с максимальной глубиной дерева 15 и показателем accuracy 0.843. Точность линейных классификаторов оказалась неудовлетворительной.